### ISSUE TITLE
successful registration, but "not-authorized" error when try to send a message

### ISSUE THREAD

**Issue by HumGuz** [tgalal/yowsup#2182]
Hi there, hope you can help me, i searched in all past issues and cant find a solution for this:

I get a succesful registration of my number and obtained the password whit no errors, but, when i try to send a mesage whit that info, i get a "no-authorized" error and i cat' n send anything, someone finded the solution?

thanks.

here is what i send to get the pass:

> python yowsup-cli registration --requestcode sms --phone 521449??????? --cc 521 --mcc 334 --mnc 020 --env android

and here is the answer:

> yowsup-cli  v2.0.15
> yowsup      v2.5.2
> 
> Copyright (c) 2012-2016 Tarek Galal
> http://www.openwhatsapp.org
> 
> This software is provided free of charge. Copying and redistribution is
> encouraged.
> 
> If you appreciate this software and you would like to support future
> development please consider donating:
> http://openwhatsapp.org/yowsup/donate
> 
> 
> INFO:yowsup.common.http.warequest:{"status":"ok","login":"521449*******","type":"existing","pw":"?????IO0QKCJyn???????PiXGlE=","expiration":4444444444.0,"kind":"free","price":"$13.00","cost":"13.00","currency":"MXN","price_expiration":1504489303}
> 
> status: ok
> kind: free
> pw: ?????IO0QKCJyn???????PiXGlE=
> price: $13.00
> price_expiration: 1504489303
> currency: MXN
> cost: 13.00
> login: 521449???????
> type: existing
> expiration: 4444444444.0

here is how i try to send my message:

> python yowsup-cli demos --config yowsup-cli.config --send 521449138???? 'Tenemos Whatsapp'

answer:

> Authentication Error: not-authorized

here is the content in my yowsup-cli.config file:

> cc=34
> phone=521449???????
> id=
> password=?????IO0QKCJyn???????PiXGlE=

**Comment by Arrheniu2** [comment]
you have to put the country code (34) in the phone number too 

cc=34
phone=34521449???????
id=
password=?????IO0QKCJyn???????PiXGlE=

**Comment by orige** [comment]
@HumGuz Did you have any success on resolve your problem? I did the same steps as you and got the same error.

**Comment by alonsodelaq** [comment]
Is there any update on this? I get the same auth error.

Anyone?

**Comment by orige** [comment]
@alonsodelaq 
Hey dude, it's working for me now.
Pay attention to these things:

1. The code that you get via sms contains dashes (888**-**123). I was registering using the code without the dashes. I was getting a password but I'm unable to sign in.

2. In Brazil, we have 9 digits besides **cc** and region number. I was trying to authenticate using the 9 digits. I don't know if this has some effect but now I'm registering numbers using the 8 digits layout

3. Use properly the mcc and mnc values

Before trying to register the number again, remove the number directory inside the ```~/.yowsup``` directory.

**Comment by alonsodelaq** [comment]
@orige 
Thanks for your answer, I tried what you suggested, I did the following:

yowsup-cli registration --requestcode sms --phone 51XXXXXXXXX --cc 51 --mcc 716 --mnc 10 -E android
yowsup-cli registration --register 123-456 --phone 51XXXXXXXXX --cc 51 -E android
yowsup-cli demos -y --login 51XXXXXXXXX:xxxxxxZMobjj0fvcW0vXnI6jTLE= -s 51zzzzzzzzz Hello

And still get the not-authorized message. I made sure I deleted the virtualenv which hade yowsup installed. Also changed the version and _MD5 parameter so I wouldn't get the "old_version" error.

Any other ideas? =/
